### DISCUSSION TITLE
Schema Validation Just Stopped Working

### DISCUSSION THREAD

**Discussion by chiester** [vazco/uniforms#1256] [Q&A] [answered]
Any idea why schema validation would just stop working? 

Async validation via `onValidate` works. But violations of the schema do not trigger `<ErrorField>` to render.

**Comment by chiester** [comment]
Also, it's still not clear to me how to get error or validation messages back to ErrorField from either sync or async validation to present in the UI.

**Comment by chiester** [comment]
If I remove `onValidate`, the schema validation works. When I put it back in, schema vaildation does not work at all, even if all the onValidate function does is return null.

**Comment by radekmie** [comment, marked as answer]
Hi @chiester. I think the `onValidate` part got answered in https://github.com/vazco/uniforms/issues/1257, and this part is clear (if not, let's continue there).

When it comes to `ErrorField`, it displays the errors using the bridge's `getError`, i.e., it extracts the field error out of the form error. To make your custom error renders in there, you'd need to make sure the error matches the format of your bridge's errors. For example, [the JSON Schema bridge works using the `details` field](https://github.com/vazco/uniforms/blob/cf3d3da51dff980f677bef26c32850dc0245b689/packages/uniforms-bridge-json-schema/src/JSONSchemaBridge.ts#L123-L145).

_(Side note: if possible, please edit the first message rather than adding multiple ones, as it's easier to follow the threads then.)_
